### PR TITLE
Use episode batch in tf/npo

### PR DESCRIPTION
**Pull request by yeukfu**
Partly address https://github.com/rlworkgroup/garage/issues/1110.

### CODE REVIEW

File: src/garage/tf/algos/npo.py [hunk 1]
```diff
@@ -474,53 +442,64 @@ def _build_entropy_term(self, i):
...
+        aug_returns = pad_tensor_n(aug_returns, self.max_episode_length)
 
         # Fit baseline
         logger.log('Fitting baseline...')
         self._baseline.fit(paths)
+        return aug_returns
 
-    def _policy_opt_input_values(self, samples_data):
-        """Map episode samples to the policy optimizer inputs.
+    def _process_episodes_batch(self, episodes):
```
Comment (nicolengsy) [inline src/garage/tf/algos/npo.py:479]: Rename to `_process_episode_batch` to keep naming consistent

**Review by nicolengsy — APPROVED** [review]

File: src/garage/tf/algos/npo.py [hunk 2]
```diff
@@ -474,53 +442,64 @@ def _build_entropy_term(self, i):
...
+                baselines.
 
         """
+        paths = episodes.to_list()
+        for i, _ in enumerate(paths):
+            paths[i]['actions'] = self._env_spec.action_space.flatten_n(
+                paths[i]['actions'])
+
+        baselines = [self._baseline.predict(path) for path in paths]
+        samples_data = paths_to_tensors(paths, self.max_episode_length,
```
Comment (ryanjulian) [inline src/garage/tf/algos/npo.py:499]: is there any way we can pass EpisodeBatch + a few other tensors to TensorFlow? A good detector that #1110 is done is "we can delete all usages of `paths_to_tensors`"

Comment (yeukfu) [reply src/garage/tf/algos/npo.py:499]: I think if padding the tensor is necessary, we must split the EpisodeBatch.

Comment (yeukfu) [reply src/garage/tf/algos/npo.py:499]: Or we can do the padding operation in TensorFlow.

File: src/garage/tf/algos/npo.py [hunk 3]
```diff
@@ -177,76 +177,45 @@ def train(self, trainer):
         last_return = None
 
         for _ in trainer.step_epochs():
-            trainer.step_path = trainer.obtain_samples(trainer.step_itr)
-            last_return = self._train_once(trainer.step_itr, trainer.step_path)
+            episodes = trainer.obtain_episodes(trainer.step_itr)
```
Comment (krzentner) [inline src/garage/tf/algos/npo.py:180]: You should still save them to `trainer.step_path`, even though they're episodes. This is needed for `save_paths` to work.

**Review by krzentner — COMMENTED** [review]
I think this change is an improvement. Honestly, I'm not sure how much of this refactoring we want to do for TF algorithms.

File: src/garage/tf/algos/npo.py (hunk 2 above)
Comment (yeukfu) [reply src/garage/tf/algos/npo.py:499]: @ryanjulian Can we do the padding operation in EpisodeBatch. For example, `episode_batch.padded_list()`. I have tried to do all things in tensorflow, but it is difficult as the padding size is a variable.

**Review by krzentner — APPROVED** [review]
Nice! Looks good to me.

**Pull request merged**